src/lib/homestore_backend/hs_homeobject.hpp

```diff
@@ -72,6 +73,24 @@ class HSHomeObject : public HomeObjectImpl {
         homestore::uuid_t index_table_uuid;
         blob_id_t blob_sequence_num;
         pg_members members[1]; // ISO C++ forbids zero-size array
+
+        uint32_t size() const { return sizeof(pg_info_superblk) + ((num_members - 1) * sizeof(pg_members)); }
+        static std::string name() { return _pg_meta_name; }
+
+        pg_info_superblk() = default;
+        pg_info_superblk(pg_info_superblk const& rhs) { *this = rhs; }
+
+        pg_info_superblk& operator=(pg_info_superblk const& rhs) {
+            id = rhs.id;
+            num_members = rhs.num_members;
+            replica_set_uuid = rhs.replica_set_uuid;
+            index_table_uuid = rhs.index_table_uuid;
+            blob_sequence_num = rhs.blob_sequence_num;
+            memcpy(members, rhs.members, sizeof(pg_members) * num_members);
```

Comment:
is this safe to copy `sizeof(pg_members) * num_members` bytes to `members`, which can hold only one `pg_members`?

Reply:
I think the idea behind the original author is that members is always come from the heap, and its pointed size is always the size of num_members.